### PR TITLE
OSD-9241 add support for ZONE_ID deployment param

### PR DESCRIPTION
**Pull request by yithian**

### CODE REVIEW

File: hack/olm-registry/olm-artifacts-template.yaml [hunk 1]
```diff
@@ -11,7 +11,13 @@ parameters:
 - name: CHANNEL
   value: staging
 - name: FEDRAMP
+<<<<<<< HEAD
```
Comment (dustman9000) [inline hack/olm-registry/olm-artifacts-template.yaml:14]: Looks like a merge conflict

Comment (yithian) [reply hack/olm-registry/olm-artifacts-template.yaml]: whoops, good catch, thanks

File: deploy/operator.yaml [hunk 2]
```diff
@@ -32,3 +32,5 @@ spec:
               value: "rh-api"
             - name: FEDRAMP
               value: "false"
+            - name: ZONE_ID
```
Comment (dustman9000) [inline deploy/operator.yaml:35]: Let's rename this to HOSTED_ZONE_ID to avoid confusion...

File: hack/olm-registry/olm-artifacts-template.yaml [hunk 3]
```diff
@@ -45,3 +47,5 @@ objects:
       env:
       - name: FEDRAMP
         value: "${FEDRAMP}"
+      - name: HOSTED_ZONE_ID
+        value: ${ZONE_ID}
```
Comment (dustman9000) [inline hack/olm-registry/olm-artifacts-template.yaml:51]: Should be HOSTED_ZONE_ID
```suggestion
        value: ${HOSTED_ZONE_ID}
```

File: pkg/controller/certificaterequest/certificaterequest_controller.go [hunk 4]
```diff
@@ -126,6 +126,12 @@ func (r *ReconcileCertificateRequest) Reconcile(request reconcile.Request) (reco
 		reqLogger.Info("running in FedRAMP environment: %b", fedramp)
 	}
 
+	var fedramp_hosted_zone_id string
+	if fedramp {
+		fedramp_hosted_zone_id = os.Getenv("HOSTED_ZONE_ID")
+		reqLogger.Info("running in FedRAMP zone: %s", fedramp_hosted_zone_id)
```
Comment (dustman9000) [inline pkg/controller/certificaterequest/certificaterequest_controller.go:132]: Check for zero length string?

**Review by dustman9000 — CHANGES_REQUESTED** [review]
Need to fix the parameter expansion from ZONE_ID to HOSTED_ZONE_ID

**Pull request merged**